File: src/Grimly.Core/Models/AppSettings.cs
using System.Text.Json.Serialization;

namespace Grimly.Models;

public sealed class AppSettings
{
    [JsonPropertyName("hotkey_modifiers")]
    public string HotkeyModifiers { get; set; } = "Ctrl+Alt";

    [JsonPropertyName("hotkey_key")]
    public string HotkeyKey { get; set; } = "G";

    [JsonPropertyName("foundry_endpoint")]
    public string FoundryEndpoint { get; set; } = "http://127.0.0.1:51318";

    /// <summary>
    /// Factory default Foundry model, as a named constant so startup logic
    /// can distinguish "user never picked a model" from a deliberate choice.
    /// </summary>
    public const string FactoryDefaultModel = "qwen2.5-7b-instruct-qnn-npu:2";

    [JsonPropertyName("model_name")]
    public string ModelName { get; set; } = FactoryDefaultModel;

    [JsonPropertyName("default_mode")]
    public EditingMode DefaultMode { get; set; } = EditingMode.FixGrammar;

    [JsonPropertyName("creativity")]
    public double Creativity { get; set; } = 0.5; // 0=precise, 0.5=default, 1=varied

    [JsonPropertyName("max_tokens")]
    public int MaxTokens { get; set; } = 2048;

    [JsonPropertyName("popup_opacity")]
    public double PopupOpacity { get; set; } = 0.95;

    [JsonPropertyName("show_floating_icon")]
    public bool ShowFloatingIcon { get; set; } = true;

    /// <summary>
    /// True once the app has auto-selected the Windows AI (Aion) model on a
    /// compatible machine. The flip happens at most once — after that the
    /// user's picker choice (including switching back to a Foundry model)
    /// is always respected.
    /// </summary>
    [JsonPropertyName("windows_ai_defaulted")]
    public bool WindowsAiDefaulted { get; set; }
}

File: mac/Grimly/Models/AppSettings.swift
import Foundation

struct AppSettings: Codable {
    var hotkeyModifiers: String = "Cmd+Option"
    var hotkeyKey: String = "G"
    var foundryEndpoint: String = "http://127.0.0.1:51318"
    var modelName: String = "qwen2.5-7b"
    var defaultMode: EditingMode = .fixGrammar
    var creativity: Double = 0.5
    var maxTokens: Int = 2048
    var popupOpacity: Double = 0.95
    var showFloatingIcon: Bool = true

    enum CodingKeys: String, CodingKey {
        case hotkeyModifiers = "hotkey_modifiers"
        case hotkeyKey = "hotkey_key"
        case foundryEndpoint = "foundry_endpoint"
        case modelName = "model_name"
        case defaultMode = "default_mode"
        case creativity
        case maxTokens = "max_tokens"
        case popupOpacity = "popup_opacity"
        case showFloatingIcon = "show_floating_icon"
    }
}
